### ISSUE TITLE
Function write_part_xbox not implemented

### ISSUE THREAD

**Issue by tlaurion** [cgsecurity/testdisk#80]
Why is that @crrodriguez @holodnak @tobiaspal @mmuman @Enlik ?

Trying to salvage drive content with testdisk recognizing all the partition table and being ready to save content but unfortunately `Function write_part_xbox not implemented` is in the way :|
![TestDiskFatX](https://user-images.githubusercontent.com/827570/82758885-39a65700-9db7-11ea-91c4-529a667685fc.png)
![TestdiskFatX_Save](https://user-images.githubusercontent.com/827570/82758888-3d39de00-9db7-11ea-9827-3aa81deb3a24.png)
![TestDiskFatX_Write_unsupported](https://user-images.githubusercontent.com/827570/82758892-432fbf00-9db7-11ea-8f53-855187ea9fa8.png)

**Comment by tlaurion** [comment]
Version: `debian buster 7.0-3+b4`

```
Get:1 https://deb.debian.org/debian buster/main amd64 testdisk amd64 7.0-3+b4 [388 kB]
Fetched 388 kB in 4s (96.9 kB/s)   
Selecting previously unselected package testdisk.
(Reading database ... 222928 files and directories currently installed.)
Preparing to unpack .../testdisk_7.0-3+b4_amd64.deb ...
Unpacking testdisk (7.0-3+b4) ...
Setting up testdisk (7.0-3+b4) ...
Processing triggers for man-db (2.8.5-2) ...

```

**Comment by tlaurion** [comment]
Same built from 8d54e20096ec347f0f144d5e0c02b32d24d374ab
`
TestDisk 7.2-WIP, Data Recovery Utility, May 2020`

**Comment by tlaurion** [comment]
Removing lines in godmode.c results in a write being permitted, but a write error.

**Comment by tlaurion** [comment]
Please tag me in if you have any useful input!

**Comment by mmuman** [comment]
Because nobody had the time and/or the need to have it implemented yet, hence the error.

**Comment by tlaurion** [comment]
@mmuman any guidelines?

**Comment by mmuman** [comment]
Not really…
You'll want to check how other partitioning schemes are implemented, and read documentation about the xbox one.
And test your code against a blank image or device you can try in something else that can read it to make sure it works.

**Comment by cgsecurity** [comment]
Fixed offsets are used for XBOX partition "table". In practice, there is no table:
https://www.jekkt.com/talks/xbox-presentation.pdf

**Comment by tlaurion** [comment]
@cgsecurity yeah, discovered fatx partitions are right..wished I could just write what was found back to disk. Have a backup I could easily write back and test.
